rest/server_context.go

```diff
@@ -367,7 +367,8 @@ func GetBucketSpec(config *DatabaseConfig, serverConfig *StartupConfig) (spec ba
 // Adds a database to the ServerContext.  Attempts a read after it gets the write
 // lock to see if it's already been added by another process. If so, returns either the
 // existing DatabaseContext or an error based on the useExisting flag.
-func (sc *ServerContext) _getOrAddDatabaseFromConfig(config DatabaseConfig, useExisting, failFast bool) (*db.DatabaseContext, error) {
+// Pass in a connectFn to replace the default ConnectToBucket function. This will cause the failFast argument to be ignored
```

Comment:
Maybe rename this function for clarity - `connectFn` implies you need to actually set up a connection.

```suggestion
// Pass in a bucketFromBucketSpecFn to replace the default ConnectToBucket function. This will cause the failFast argument to be ignored
```

Reply:
Changed `AddDatabaseFromConfigWithConnectFn` to `AddDatabaseFromConfigWithBucket` and made it take a bucket now instead then pass that down using a `bucketFromBucketSpecFn` function.